cookbook/controller/argument_value_resolver.rst

```diff
@@ -0,0 +1,163 @@
+    {
+        public function indexAction(User $user)
+        {
+            return new Response('<html><body>Hello '.$user->getUsername().'!</body></html>');
+        }
+    }
+
+Somehow you will have to get the ``User`` object and inject it into our action. This can be done
+by implementing the :class:`Symfony\\Component\\HttpKernel\\Controller\\ArgumentValueResolverInterface`.
+This interface specifies that you have to implement two methods::
+
+    interface ArgumentValueResolverInterface
+    {
+        public function supports(Request $request, ArgumentMetadata $argument);
+        public function resolve(Request $request, ArgumentMetadata $argument);
+    }
+
+  * The ``supports()`` method is used to check whether the resolver supports the given argument. It will
+    only continue if it returns ``true``.
+
+  * The ``resolve()`` method will be used to resolve the actual value just acknowledged by
+    ``supports()``. Once a value is resolved you can ``yield`` the value to the ``ArgumentResolver``.
+
+  * The ``Request`` object is the current ``Request`` which would also be injected into your
+    action in the forementioned functionality.
+
+  * The :class:``Symfony\\Component\\HttpKernel\\ControllerMetadata\\ArgumentMetadata`` represents
+    information retrieved from the method signature for the current argument it's trying to resolve.
+
+.. note::
+
+    The ``ArgumentMetadata`` is a simple data container created by the
+    :class:``Symfony\\Component\\HttpKernel\\ControllerMetadata\\ArgumentMetadataFactory``. This
+    factory will work on every supported PHP version but might give different results. E.g. the
+    ``isVariadic()`` will never return true on PHP 5.5 and only on PHP 7.0 and higher it will give
+    you basic types when calling ``getType()``.
+
+Now that you know what to do, you can implement this interface. In order to get the current ``User``,
+you will have to get it from the ``TokenInterface`` which is in the ``TokenStorageInterface``::
+
+    namespace AppBundle\ArgumentValueResolver;
+
+    use AppBundle\User;
+    use Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface;
+    use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+    class UserValueResolver implements ArgumentValueResolverInterface
+    {
+        private $tokenStorage;
+
+        public function __construct(TokenStorageInterface $tokenStorage)
+        {
+            $this->tokenStorage = $tokenStorage;
+        }
+
+        public function supports(Request $request, ArgumentMetadata $argument)
+        {
+            return ($token = $this->tokenStorage->getToken()) && $token->getUser() instanceof User;
+        }
+
+        public function resolve(Request $request, ArgumentMetadata $argument)
+        {
+            yield $this->tokenStorage->getToken()->getUser();
+        }
+    }
+
+This was pretty simple, now all you have to do is add the configuration for the service container. This
```

Comment:
Some people don't like using "simple" etc. in documentations, as things can be a lot harder for beginners than you imagine. Maybe replace "This was pretty simple" with "That's it!" to be a bit more neutral.